### ISSUE TITLE
Null Pointer Exception when server connection closed and onError sequence is not triggered

### ISSUE THREAD

**Issue by rosensilva** [wso2/product-ei#5291]
**Description:**
Null pointer exception is thrown when the remote connection is closed by the server prematurely before sending a response to EI. Their onError sequence is not getting triggering because of this error.

Following is the stack trace.
```
[2020-08-25 19:18:05,763] [-1] [] [HTTPS-Sender I/O dispatcher-2]  WARN {org.apache.synapse.transport.passthru.TargetHandler} -  Connection closed by target host before receiving the response  Remote Address : bor58.dayforcehcm.com/45.60.121.205:443
[2020-08-25 19:18:05,776] [-1234] [] [PassThroughMessageProcessor-139] ERROR {org.apache.axis2.transport.base.threads.NativeWorkerPool} -  Uncaught exception endpoints.EndpointDefinition}TID: [-1234] [] 
[2020-08-26 11:30:01,701]  WARN {org.apache.synapse.endpoints.EndpointDefinition} -  Error while evaluating dynamic endpoint timeout expression.Synapse global timeout is taken as effective timeout. {org.apache.synapse.endpoints.EndpointDefinition}TID: [-1] [] 
[2020-08-26 11:32:02,430]  WARN {org.apache.synapse.transport.passthru.TargetHandler} -  Connection closed by target host before receiving the response  Remote Address : bor58.dayforcehcm.com/45.60.121.205:443 {org.apache.synapse.transport.passthru.TargetHandler}TID: [-1234] [] 
[2020-08-26 11:32:02,440] ERROR {org.apache.axis2.transport.base.threads.NativeWorkerPool} -  Uncaught exception {org.apache.axis2.transport.base.threads.NativeWorkerPool}java.lang.NullPointerException        
at org.apache.synapse.core.axis2.SynapseCallbackReceiver.handleMessage(SynapseCallbackReceiver.java:264)        
at org.apache.synapse.core.axis2.SynapseCallbackReceiver.receive(SynapseCallbackReceiver.java:195)        
at org.apache.synapse.transport.passthru.TargetErrorHandler$1.run(TargetErrorHandler.java:140)        
at org.apache.axis2.transport.base.threads.NativeWorkerPool$1.run(NativeWorkerPool.java:172)        
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)        
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)        
at java.lang.Thread.run(Thread.java:748)
```

**Steps to reproduce:**

1. Create a backend that has a small socket timeout. We have used a separate EI with a port offset and changed the http.socket.timeout as 15000 in the passthru-http.properties file and deployed the backend proxy. be.xml
2. Use EI 6.5.0 pack and deployed the below proxy and the template mediator. prox_1.0.0.car tem_1.0.0.car This proxy invokes the template mediator and the template mediator invokes the backend we created.
Scheduled a task to invoke the proxy. LMSExtract_TASK-1.0.0.xml
Then you will be able to reproduce the exact error.
[artifacts.zip](https://github.com/wso2/product-ei/files/5401781/artifacts.zip)

**Comment by oscarvesga** [comment]
Hello @rosensilva, Do you know in what integrator version this fix is implemented? I am using the docker-hub GA version (v6.5.0)

**Comment by rosensilva** [comment]
Hi @oscarvesga this is fixed in Micro Integrator 4.0.0

**Comment by oscarvesga** [comment]
Do you have any documentation about how I can upgrade from integrator 6.5.0 to microintegrator 4.0.0?
